src/lib/features/project/project-service.ts

```diff
@@ -214,16 +219,18 @@ export default class ProjectService {
         this.isEnterprise = config.isEnterprise;
         this.resourceLimits = config.resourceLimits;
         this.eventBus = config.eventBus;
+        this.projectReadModel = projectReadModel;
     }
 
     async getProjects(
         query?: IProjectQuery,
         userId?: number,
-    ): Promise<IProjectWithCount[]> {
-        const projects = await this.projectStore.getProjectsWithCounts(
-            query,
-            userId,
-        );
+    ): Promise<TransitionalProjectData[]> {
+        const getProjects = this.flagResolver.isEnabled('useProjectReadModel')
+            ? this.projectReadModel.getProjectsForAdminUi
+            : this.projectStore.getProjectsWithCounts;
+
+        const projects = await getProjects(query, userId);
 
```

Comment:
This is the main change. With feature flag off, everything will work as before. With flag on, use the new read model instead.